### PR TITLE
Add `outline_opacity` to `add_plane_widget`

### LINKED ISSUE #5353: Remove bound box in p.add_mesh_slice function
### Discussed in https://github.com/pyvista/pyvista/discussions/5346

<div type='discussions-op-text'>

<sup>Originally posted by **SantiagoPendon** December 15, 2023</sup>
Hi to everyone once again :)

Following the next tutorial https://docs.pyvista.org/version/stable/examples/03-widgets/plane-widget.html#sphx-glr-examples-03-widgets-plane-widget-py
I'm very impressed with the results, but I would like to know if is possible remove the black box cube. The arrow to displace the volume and slicer is great, but the black box is unnecessary for my intentions. 

Thanks so much!</div>

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

resolve #5353 

### Details

- None

### CODE REVIEW

File: pyvista/plotting/widgets.py [hunk 1]
```diff
@@ -549,6 +556,9 @@ def _the_callback(widget, _event):
             if not normal_rotation:
                 plane_widget.GetNormalProperty().SetOpacity(0)
 
+            if not outline_visible:
+                plane_widget.GetOutlineProperty().SetOpacity(0)
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:560]: This method accepts a double between 0 and 1.  Should we instead make this a `outline_opacity`?  We could also consider adding in a dict for any property to be set.

Comment (tkoyama010) [reply pyvista/plotting/widgets.py]: resolved in 40e30759499e13c3658a6d465b831f6114724c8f

File: pyvista/plotting/widgets.py [hunk 2]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
             .. versionchanged:: 0.38.0
                Now accepts either strings and ``vtk.vtkCommand.EventIds``.
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:469]: We should also tell the user they can use a float value.  I'm okay if we also accept a Boolean as this converts correctly to the behavior.

File: pyvista/plotting/widgets.py [hunk 3]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
...
 
+        outline_opacity : bool, optional
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:470]:
```suggestion
        outline_opacity : bool or float, optional
```

File: pyvista/plotting/widgets.py [hunk 4]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
...
+            Set the visible of outline. Only valid when using
+            an implicit plane.
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:472]:
```suggestion
            an implicit plane. Either a string or float.
```

File: pyvista/plotting/widgets.py [hunk 5]
```diff
@@ -704,6 +714,12 @@ def add_mesh_clip_plane(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool, optional
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:717]:
```suggestion
        outline_opacity : bool or float, optional
```

File: pyvista/plotting/widgets.py [hunk 6]
```diff
@@ -704,6 +714,12 @@ def add_mesh_clip_plane(
...
+            Set the visible of outline. Only valid when using
+            an implicit plane.
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:719]:
```suggestion
            an implicit plane.
            Either a string or float.
```

File: pyvista/plotting/widgets.py [hunk 7]
```diff
@@ -865,6 +883,12 @@ def add_volume_clip_plane(
...
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:888]:
```suggestion
            an implicit plane. Either a string or float.
```

File: pyvista/plotting/widgets.py [hunk 8]
```diff
@@ -865,6 +883,12 @@ def add_volume_clip_plane(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool, optional
```
Comment (tkoyama010) [inline pyvista/plotting/widgets.py:886]:
```suggestion
        outline_opacity : bool or float, optional
```

File: pyvista/plotting/widgets.py (hunk 2 above)
Comment (tkoyama010) [reply pyvista/plotting/widgets.py:469]: Thanks a lot! resolved in 48405a1d99333525fd6fed7c7cc399aded6d2761 1aceb6f7eed7e415650c267ba5ec881e1905af04 02ca44414de704c50e070e5c2adff77e399eafef .

File: pyvista/plotting/widgets.py [hunk 9]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
             .. versionchanged:: 0.38.0
                Now accepts either strings and ``vtk.vtkCommand.EventIds``.
 
+        outline_opacity : bool or float, optional
+            Set the visible of outline. Only valid when using
+            an implicit plane. Either a string or float.
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:472]:
```suggestion
            an implicit plane. Either a bool or float.
```

File: pyvista/plotting/widgets.py [hunk 10]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
...
+
+            .. versionchanged:: 0.44.0
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:474]:
```suggestion
            .. versionadded:: 0.44.0
```

File: pyvista/plotting/widgets.py [hunk 11]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
             .. versionchanged:: 0.38.0
                Now accepts either strings and ``vtk.vtkCommand.EventIds``.
 
+        outline_opacity : bool or float, optional
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:470]:
```suggestion
        outline_opacity : bool or float, default: True
```

File: pyvista/plotting/widgets.py [hunk 12]
```diff
@@ -704,6 +714,12 @@ def add_mesh_clip_plane(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool or float, optional
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:717]:
```suggestion
        outline_opacity : bool or float, default=True
```

File: pyvista/plotting/widgets.py [hunk 13]
```diff
@@ -704,6 +714,12 @@ def add_mesh_clip_plane(
...
+            Set the visible of outline. Only valid when using
+            an implicit plane. Either a string or float.
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:719]:
```suggestion
            an implicit plane. Either a bool or float.
```

File: pyvista/plotting/widgets.py [hunk 14]
```diff
@@ -704,6 +714,12 @@ def add_mesh_clip_plane(
...
+
+            .. versionchanged:: 0.44.0
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:721]:
```suggestion
            .. versionadded:: 0.44.0
```

File: pyvista/plotting/widgets.py [hunk 15]
```diff
@@ -865,6 +883,12 @@ def add_volume_clip_plane(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool or float, optional
+            Set the visible of outline. Only valid when using
+            an implicit plane. Either a string or float.
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:888]:
```suggestion
            an implicit plane. Either a bool or float.
```

File: pyvista/plotting/widgets.py [hunk 16]
```diff
@@ -865,6 +883,12 @@ def add_volume_clip_plane(
...
+
+            .. versionchanged:: 0.44.0
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:890]:
```suggestion
            .. versionadded:: 0.44.0
```

File: pyvista/plotting/widgets.py [hunk 17]
```diff
@@ -989,6 +1015,12 @@ def add_mesh_slice(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool or float, optional
+            Set the visible of outline. Only valid when using
+            an implicit plane. Either a string or float.
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:1020]:
```suggestion
            an implicit plane. Either a bool or float.
```

File: pyvista/plotting/widgets.py [hunk 18]
```diff
@@ -989,6 +1015,12 @@ def add_mesh_slice(
...
+
+            .. versionchanged:: 0.44.0
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:1022]:
```suggestion
            .. versionadded:: 0.44.0
```

File: pyvista/plotting/widgets.py [hunk 19]
```diff
@@ -989,6 +1015,12 @@ def add_mesh_slice(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool or float, optional
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:1018]:
```suggestion
        outline_opacity : bool or float, default: True
```

**Comment by tkoyama010** [comment]
Thanks for pointing out.

**Comment by tkoyama010** [comment]
I set default to `None` for backwards compatibility in case `opacity` is set in `color`.

File: pyvista/plotting/widgets.py [hunk 20]
```diff
@@ -466,6 +467,12 @@ def add_plane_widget(
             .. versionchanged:: 0.38.0
                Now accepts either strings and ``vtk.vtkCommand.EventIds``.
 
+        outline_opacity : bool or float
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:470]:
```suggestion
        outline_opacity : bool or float, optional
```

File: pyvista/plotting/widgets.py [hunk 21]
```diff
@@ -704,6 +715,12 @@ def add_mesh_clip_plane(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool or float, default=True
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:718]:
```suggestion
        outline_opacity : bool or float, optional
```

File: pyvista/plotting/widgets.py [hunk 22]
```diff
@@ -989,6 +1016,12 @@ def add_mesh_slice(
         origin : tuple(float), optional
             The starting coordinate of the center of the plane.
 
+        outline_opacity : bool or float, default: True
```
Comment (MatthewFlamm) [inline pyvista/plotting/widgets.py:1019]:
```suggestion
        outline_opacity : bool or float, optional
```

**Review by MatthewFlamm — COMMENTED** [review]
Good catch with the color usage.  Now need to change back to `optional`.

**Review by MatthewFlamm — APPROVED** [review]
LGTM

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

**Pull request merged**